self_balancing_tree.py

```diff
@@ -0,0 +1,128 @@
+# Hacker rank problem link: https://www.hackerrank.com/challenges/self-balancing-tree
+
+
+class Node:
+    def __init__(self,data , left=None, right=None):
+        self.data = data
+        self.height = 0
+
+
+def height(root):

```

Comment:
This should be a function of Node, not a free function. A free function is a function that's not attached to a class (it's a C++ism - https://stackoverflow.com/questions/4861914/what-is-the-meaning-of-the-term-free-function-in-c - but it's commonly used for other languages).